src/wp-includes/media.php

```diff
@@ -5614,3 +5855,21 @@ function wp_increase_content_media_count( $amount = 1 ) {
 
 	return $content_media_count;
 }
+
+/**
+ * If the media is a possible candidate for fetchpriority='high'.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param bool $value Optional. Used to change the static variable. Default null.
+ * @return bool $maybe_fetchpriority_high_media Return true if no other LCP candidate image found yet else false.
+ */
+function wp_maybe_fetchpriority_high_media( $value = null ) {
```

Comment:
Initially, I used only a static variable inside `wp_get_loading_optimization_attributes` function but then had to shift to this function as some of the test-cases need to reset this value. 

cc: @felixarntz  @joemcgill

Reply:
This make sense. 
However, even with this new function i'll still need to add an additional function for the flagged static variable.

W.r.t test cases we need the ability to reset the static variable back to true, i can only think of a function to do that. ( noting that this reseting logic is only needed for test cases and would never ever be needed in a live scenario ). Do let me know if there can be a better alternative. cc: @felixarntz

Reply:
@kt-12 Fair point, in that case let's go with a very specific function similar to `wp_increase_content_media_count()`. Maybe something like `wp_set_high_priority_element_rendered_flag()`?